### PR TITLE
Add python3 support

### PR DESCRIPTION
**Pull request by cedeyn**
* Fix last python3 specifics
* Add ci tests through travis

### CODE REVIEW

File: lib/MilkCheck/Engine/BaseEntity.py [hunk 1]
```diff
@@ -421,13 +421,13 @@ def remove_dep(self, dep_name, parent=True):
 
     def clear_parent_deps(self):
         '''Remove all parent dependencies of an entity'''
-        for dpname in self.parents.keys():
-            self.remove_dep(dpname)
+        dpnames = self.parents.copy().keys()
```
Comment (degremont) [inline lib/MilkCheck/Engine/BaseEntity.py]: probably easier to read to just add a `list()` call

    for dpname in list(self.parents.keys()):

File: lib/MilkCheck/Engine/BaseEntity.py [hunk 2]
```diff
@@ -421,13 +421,13 @@ def remove_dep(self, dep_name, parent=True):
...
+        dpnames = [self.remove_dep(dpname) for dpname in dpnames]
 
     def clear_child_deps(self):
         '''Remove all child dependencies of an entity'''
-        for dpname in self.children.keys():
-            self.remove_dep(dep_name=dpname, parent=False)
+        dpnames = self.children.copy().keys()
```
Comment (degremont) [inline lib/MilkCheck/Engine/BaseEntity.py]: same here

File: lib/MilkCheck/Engine/ServiceGroup.py [hunk 3]
```diff
@@ -192,7 +192,7 @@ def __update_edges(self, create_links=False):
                     deps = abs_ser.parents
                 else:
                     deps = abs_ser.children
-                for dep in deps.values():
+                for dep in deps.copy().values():
```
Comment (degremont) [inline lib/MilkCheck/Engine/ServiceGroup.py]: prefer list()

File: lib/MilkCheck/Engine/ServiceGroup.py [hunk 4]
```diff
@@ -211,9 +211,9 @@ def remove_inter_dep(self, dep_name):
         if not self.has_subservice(dep_name):
             raise ServiceNotFoundError()
         else:
-            for dep in self._subservices[dep_name].parents.values():
+            for dep in self._subservices[dep_name].parents.copy().values():
```
Comment (degremont) [inline lib/MilkCheck/Engine/ServiceGroup.py]: same here and below

File: lib/MilkCheck/Engine/BaseEntity.py (hunk 1 above)
Comment (cedeyn) [reply lib/MilkCheck/Engine/BaseEntity.py]: Done

File: lib/MilkCheck/Engine/BaseEntity.py (hunk 2 above)
Comment (cedeyn) [reply lib/MilkCheck/Engine/BaseEntity.py]: Done

File: lib/MilkCheck/Engine/ServiceGroup.py (hunk 3 above)
Comment (cedeyn) [reply lib/MilkCheck/Engine/ServiceGroup.py]: Done

File: lib/MilkCheck/Engine/ServiceGroup.py (hunk 4 above)
Comment (cedeyn) [reply lib/MilkCheck/Engine/ServiceGroup.py]: Done.

I did the same for ServiceManager also.

File: lib/MilkCheck/Engine/BaseEntity.py [hunk 5]
```diff
@@ -421,13 +421,13 @@ def remove_dep(self, dep_name, parent=True):
 
     def clear_parent_deps(self):
         '''Remove all parent dependencies of an entity'''
-        for dpname in self.parents.keys():
-            self.remove_dep(dpname)
+        dpnames = list(self.parents.keys())
```
Comment (degremont) [inline lib/MilkCheck/Engine/BaseEntity.py]: I mean keeping the original for-loop, just adding a `list()` call.

        for dpname in list(self.parents.keys()):
            self.remove_dep(dpname)

same below :)

**Comment by degremont** [comment]
one last change, otherwise LGTM

File: lib/MilkCheck/Engine/BaseEntity.py (hunk 5 above)
Comment (cedeyn) [reply lib/MilkCheck/Engine/BaseEntity.py]: You're right ;)
Should be ok now.

**Pull request merged**